cd/build_images.sh

```diff
@@ -18,31 +18,36 @@
 # Required for variables from sourced env file to automatically be visible to docker compose.
 set -e -a
 
-root_directory="$( cd $(dirname $(dirname $0)) > /dev/null 2>&1 && pwd )"
-cd $root_directory
+ROOT_DIR="$(readlink -f "$(dirname "$(dirname "${0}")")")"
 
-if [ ! -z "$1" ]; then
-	ENV_FILE=$1
-else
-	echo "Error - Environment file not set"; exit 1;
-fi
+pushd "${ROOT_DIR}" || exit 1
 
-# The following command sets:
+# The following env file will be sourced to set:
 # HADOOP_VERSION
 # GAFFER_VERSION
 # GAFFERPY_VERSION
 # ACCUMULO_VERSION
 # SPARK_VERSION
-source "${ENV_FILE}"
+# TINKERPOP_VERSION
+if [[ -f "${1}" ]]; then
+    source "${1}"
+else
+    echo "Error - Environment file not set"
+    exit 1
+fi
 
 # Builds all of the Gaffer and Accumulo related images:
 docker compose --project-directory ./docker/accumulo/ -f ./docker/accumulo/docker-compose.yaml build
 docker compose --project-directory ./docker/gaffer-road-traffic-loader/ -f ./docker/gaffer-road-traffic-loader/docker-compose.yaml build
 # Builds all of the notebook related images:
 docker compose --project-directory ./docker/gaffer-pyspark-notebook/ -f ./docker/gaffer-pyspark-notebook/docker-compose.yaml build notebook
 docker compose --project-directory ./docker/spark-py/ -f ./docker/spark-py/docker-compose.yaml build
+# Builds the Gaffer Gremlin server
+./docker/gaffer-gremlin/build.sh
```

Comment:
If this command is added to the compose build section it would keep consistency here and allow building via compose too.

Reply:
That would be nice but I don't think you can add a command to the compose build section yaml, you can only really edit the build context and arguments. Maven needs to be ran before the build so the Jars are available hence this script is just a handy wrapper to run maven then the compose build.

I could just run the maven command and compose command separately here but don't see much point as it would just be duplicating whats in the script.

Another option would be to move the maven command run into the Dockerfile but this would then not allow development builds where the Jars are only available locally.

Reply:
I see. I think it comes back to the comment from @t92549 around the approach of using maven outside the Dockerfile. It would be good to allow both this new approach and the previous solution, but that's something which can be looked into separately - it doesn't need to be part of this PR.

Reply:
Agreed, usually when containerising a java app the java build will be ran first (e.g. via maven) then the built Jar(s) simply copied into the image via the Dockerfile. This kind of circles back to the point I made about it being quite non standard to separate the container builds to another repo as you lose that standard approach a bit.